07-processing-vector-data.qmd

```diff
@@ -163,11 +174,46 @@ all_stack <- c(vect_stack, rast_stack)
 plot(all_stack)
 
 # we can write this out as a tif (raster object) 
-#writeRaster(all_stack, file.path("clean_data", "rstack.tif"), overwrite = T)
+#writeRaster(all_stack, "clean_data/rstack.tif", overwrite = T)
 ```
 
-Or we can write this out to a very small R object 
+Or we can write this out to a very small R object
+
 ```{r}
 #| eval: !expr '!file.exists(file.path("clean_data", "covars.RDS")) || isTRUE(params$write)'
-saveRDS(all_stack, file.path("clean_data", "covars.RDS")) # much faster
+saveRDS(all_stack, "clean_data/covars.RDS") # much faster
+```
+
+## Your turn
+
+::: {.callout-tip collapse="true"}
+1.  Generate a road density later using a different function (i.e. not sum), and or different scale of window (w). Hint: use the ?focal to see other options available.
+
+Solution:
+
+```{r, eval = FALSE}
+# One example would be to generate the "max" value of the moving window. Note the w parameter is also reduced in the example. 
+
+rrdensmax <- focal(rroads, w=3, fun="max", na.rm=TRUE)
+
+```
+:::
```

Comment:
I think you want the question outside the callout

````suggestion
1.  Generate a road density later using a different function (i.e. not sum), and or different scale of window (w). Hint: use the ?focal to see other options available.

::: {.callout-tip collapse="true"}
Solution:

```{r, eval = FALSE}
# One example would be to generate the "max" value of the moving window. Note the w parameter is also reduced in the example. 

rrdensmax <- focal(rroads, w=3, fun="max", na.rm=TRUE)

```
:::
````